kernel/init.c

```diff
@@ -508,46 +508,38 @@ static FUNC_NORETURN void switch_to_main_thread(char *stack_ptr)
 }
 #endif /* CONFIG_MULTITHREADING */
 
-#if defined(CONFIG_ENTROPY_HAS_DRIVER) || defined(CONFIG_TEST_RANDOM_GENERATOR)
 __boot_func
 void __weak z_early_rand_get(uint8_t *buf, size_t length)
 {
-#ifdef CONFIG_ENTROPY_HAS_DRIVER
-	const struct device *const entropy = DEVICE_DT_GET_OR_NULL(DT_CHOSEN(zephyr_entropy));
+	static uint64_t state = 123456789UL;
```

Comment:
Two minor items about this constant 123456789UL.

The first is that since _state_ is of type _uint64_t_, it makes me think that we should be using the _ULL_ suffix instead of _UL_.

The second is that I am wary about this being a hard-coded constant for the fallback case. I am imagining a scenario where there is a fleet of devices/products so predictable in their timing such that when their software invokes `z_early_rand_get()` that they all get the same results--thereby negating some of the randomness. To cover that case, we could make this _123456789_ constant configurable. Thoughts?

Reply:
Yep, I thought about it. This constant I got from the timer based random implementation, just  tried to get rid of existent issues. I also have made it as weak symbol to allow applications to come up with a better random generator if possible. Make the constant will still be very predictable but at least won't be an universal value.

Reply:
I forgot all about the weak attribute when I wrote that. That totally allows for any necessary customisation.

Reply:
@peter-mitsis even allowing replace this function, I think it is worth to add a symbol to customize it. Will send a new version doing it.